[Bug]: WAV audio-only recording is broken

### Expected behavior?

On using `File->Record->Start sound recording...` and selecting the WAV format, a WAV file is created to capture the audio.

### Actual behavior?

Everything works as expected, but the RIFF and data chunk headers are broken (the sizes are set to FFFFFFFFh). I believe this is caused by [this line](https://github.com/visualboyadvance-m/visualboyadvance-m/blob/f915ec5972eb517f0972570bd1f017b01083c1a4/src/common/ffmpeg.cpp#L417), which only calls `av_write_trailer()` when _not_ recording in audio-only mode.

Inspecting the ffmpeg source, the call to `av_write_trailer()` should invoke `wav_write_trailer()` in libavformat/wavenc.c, which then writes the correct chunk sizes, as they'd been set to FFFFFFFFh when `avformat_write_header()` invoked `wav_write_header()`. So in other words: the call to `av_write_trailer()` should _always_ be performed, not only when doing a video recording.

### Steps to reproduce the problem

1. Start audio recording with WAV output.
2. Stop recording.
3. Inspect the resulting file.

### Error messages?

_No response_

### What game were you playing

_No response_

### What settings are you using?

Default settings, but N/A; this is an ffmpeg interfacing issue.

### Specifications and Build details

VBA-M 2.1.8 (64-bit). Should be present in all current builds, though. Not sure how far back this issue goes, since it seems to be related to ffmpeg's API changing through the years.

### Copyright Compliance

- [X] There are no infringing files attached to this bug report.

Alright, I ran cmake inside the developer Powershell. Using cmake inside the build-msvc64 folder failed (`CMake Error: The source directory "/c/Users/Aikku/Documents/GitHub/visualboyadvance-m/build-msvc64" does not appear to contain CMakeLists.txt.`), so I moved up one directory to the root folder and ran cmake again. [I've attached the console output from doing this](https://github.com/visualboyadvance-m/visualboyadvance-m/files/14006504/CMakeErrors.txt).

Alright, I ran `cmake ../ -DVCPKG_TARGET_TRIPLET=[...]`, and then ran `cmake -DVCPKG_TARGET_TRIPLET=[...]`, both from inside the build-msvc64 directory. The warning about `No source or binary directory provided.  Both will be assumed to be [...]` doesn't display anymore, but now it stops earlier:

```
[...]
-- Downloading https://nightly.vba-m.com/vcpkg/x64-windows-static/zlib_1.3_x64-w
indows-static.zip ...
-- done.
Import-Module: The specified module '/c/Users/Aikku/Documents/GitHub/visualboyad
vance-m/vcpkg-binpkg/vcpkg-binpkg.psm1' was not loaded because no valid module f
ile was found in any module directory.
vcpkg-instpkg: The term 'vcpkg-instpkg' is not recognized as a name of a cmdlet,
 function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path
is correct and try again.
Could NOT find SFML (missing:  SFML_NETWORK_LIBRARY SFML_SYSTEM_LIBRARY)
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230
(message):
  Could NOT find OpenGL (missing: OPENGL_opengl_LIBRARY OPENGL_glx_LIBRARY
  OPENGL_INCLUDE_DIR)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILU
RE_MESSAGE)
  /usr/share/cmake/Modules/FindOpenGL.cmake:545 (FIND_PACKAGE_HANDLE_STANDARD_AR
GS)
  C:/Program Files/Microsoft Visual Studio/2022/Community/VC/vcpkg/scripts/build
systems/vcpkg.cmake:859 (_find_package)
  CMakeLists.txt:429 (find_package)
```

--- a/src/common/ffmpeg.cpp
+++ b/src/common/ffmpeg.cpp
@@ -339,34 +339,34 @@ recording::MediaRet recording::MediaRecorder::Record(const char *fname, int widt
     ret = setup_common(fname);
     if (ret != MRET_OK)
     {
-        Stop();
+        Stop(false);
         return ret;
     }
     // video stream
     ret = setup_video_stream_info(width, height, depth);
     if (ret != MRET_OK)
     {
-        Stop();
+        Stop(false);
         return ret;
     }
     ret = setup_video_stream(width, height);
     if (ret != MRET_OK)
     {
-        Stop();
+        Stop(false);
         return ret;
     }
     // audio stream
     ret = setup_audio_stream();
     if (ret != MRET_OK)
     {
-        Stop();
+        Stop(false);
         return ret;
     }
     // last details
     ret = finish_setup(fname);
     if (ret != MRET_OK)
     {
-        Stop();
+        Stop(false);
         return ret;
     }
     return MRET_OK;
@@ -409,12 +409,12 @@ recording::MediaRet recording::MediaRecorder::AddFrame(const uint8_t *vid)
     return MRET_OK;
 }
 
-void recording::MediaRecorder::Stop()
+void recording::MediaRecorder::Stop(bool initSuccess)
 {
     if (oc)
     {
         // write the trailer; must be called before av_codec_close()
-        if (!audioOnlyRecording)
+        if (initSuccess) // only call av_write_trailer() if initialization went ok
             av_write_trailer(oc);
     }
     isRecording = false;
@@ -521,21 +521,21 @@ recording::MediaRet recording::MediaRecorder::Record(const char *fname)
     ret = setup_common(fname);
     if (ret != MRET_OK)
     {
-        Stop();
+        Stop(false);
         return ret;
     }
     // audio stream
     ret = setup_audio_stream();
     if (ret != MRET_OK)
     {
-        Stop();
+        Stop(false);
         return ret;
     }
     // last details
     ret = finish_setup(fname);
     if (ret != MRET_OK)
     {
-        Stop();
+        Stop(false);
         return ret;
     }
     return MRET_OK;

--- a/src/common/ffmpeg.h
+++ b/src/common/ffmpeg.h
@@ -67,7 +67,7 @@ class MediaRecorder
         // start audio only
         MediaRet Record(const char *fname);
         // stop both
-        void Stop();
+        void Stop(bool initSuccess = true);
         bool IsRecording()
         {
                 return isRecording;

